### PR TITLE
Add about content with examples

### PR DESCRIPTION
**Pull request by jujaga**
# Description

This PR adds a descriptive about page to the frontend in order to allow users to understand and acquaint themselves with dgrsc and cdogs. It also includes QoL improvements and some UI polish.
* Add session storage caching for /config requests
* Add page loading spinner
* Add more stringent validation requirements for file input form
* Add table of example templates and contexts
* Minor CSS/Layout consistency improvements

This change is needed in order to allow users to know what this application is doing.

[[SHOWCASE-375]](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-375)
[[SHOWCASE-376]](https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-376)

## Types of changes

Bug fix (non-breaking change which fixes an issue)
New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
SonarQube Analysis: https://sonarqube-wfezkf-tools.pathfinder.gov.bc.ca/dashboard?id=document-generation-showcase-pr-17

File: app/frontend/src/views/Home.vue [hunk 1]
```diff
@@ -1,12 +1,149 @@
 <template>
   <v-container class="home">
-    <h2>Welcome to Document Generation Showcase (DGRSC).</h2>
+    <h2 class="text-center">Welcome to Document Generation Showcase (DGRSC).</h2>
```
Comment (loneil) [inline app/frontend/src/views/Home.vue]: Probably don't need a period at the end of a title

File: app/frontend/src/views/Home.vue [hunk 2]
```diff
@@ -1,12 +1,149 @@
...
+          template: {
+            color: 'orange',
+            icon: 'mdi-file-powerpoint',
+            link: './examples/template_powerpoint.pptx',
+          },
+          type: 'Microsoft PowerPoint Presentation (.pptx)'
+        },
+        {
+          contexts: {
+            color: 'red',
```
Comment (loneil) [inline app/frontend/src/views/Home.vue]: Would change to black

File: app/frontend/src/views/Home.vue [hunk 3]
```diff
@@ -1,12 +1,149 @@
...
+          template: {
+            color: 'green',
+            icon: 'mdi-file-excel',
+            link: './examples/template_movies.xlsx',
+          },
+          type: 'Microsoft Excel Worksheet (.xlsx)'
+        },
+        {
+          contexts: {
+            color: 'purple',
```
Comment (loneil) [inline app/frontend/src/views/Home.vue]: Would change to black

File: app/frontend/src/components/JWTDebug.vue [hunk 4]
```diff
@@ -0,0 +1,35 @@
+<template>
+  <v-bottom-sheet v-if="devMode" inset v-model="sheet">
```
Comment (loneil) [inline app/frontend/src/components/JWTDebug.vue]: I would put the v-if="devMode" up a component at the point where this component is added (Header.vue) so someone looking at the higher level layout knows when this component is rendered or not.

File: app/frontend/src/components/JWTDebug.vue [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    <template v-slot:activator="{ on }">
+      <v-btn outlined dark v-on="on">Debug</v-btn>
+    </template>
+    <v-sheet class="text-center">
+      <v-btn class="sheet-close ma-2" color="primary" @click="sheet = !sheet">close</v-btn>
+      <v-card class="sheet-debug pa-2 ma-2" raised>
+        <v-card-text>
+          <b
```
Comment (loneil) [inline app/frontend/src/components/JWTDebug.vue]: Should use _strong_ instead of _b_

File: app/frontend/src/components/JWTDebug.vue [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+            v-if="$keycloak.authenticated"
+          >JWT Debug - {{$keycloak.fullName}} ({{$keycloak.userName}}) is logged in.</b>
+          <b v-else>JWT Debug - Not logged in.</b>
```
Comment (loneil) [inline app/frontend/src/components/JWTDebug.vue]: strong

File: app/frontend/src/components/JWTDebug.vue [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  data() {
+    return {
+      devMode: process.env.NODE_ENV === 'development',
+      sheet: false
+    };
+  },
+};
+</script>
+
+<style scoped>
```
Comment (loneil) [inline app/frontend/src/components/JWTDebug.vue]: Empty style tags can be removed

File: app/frontend/src/views/Home.vue (hunk 1 above)
Comment (jujaga) [reply app/frontend/src/views/Home.vue]: Fixed

File: app/frontend/src/views/Home.vue (hunk 3 above)
Comment (jujaga) [reply app/frontend/src/views/Home.vue]: Made all JSON files teal for consistency.

File: app/frontend/src/views/Home.vue (hunk 2 above)
Comment (jujaga) [reply app/frontend/src/views/Home.vue]: Made all JSON files teal for consistency.

File: app/frontend/src/components/JWTDebug.vue (hunk 5 above)
Comment (jujaga) [reply app/frontend/src/components/JWTDebug.vue]: Fixed

File: app/frontend/src/components/JWTDebug.vue (hunk 7 above)
Comment (jujaga) [reply app/frontend/src/components/JWTDebug.vue]: Fixed

File: app/frontend/src/components/JWTDebug.vue (hunk 6 above)
Comment (jujaga) [reply app/frontend/src/components/JWTDebug.vue]: Fixed

File: app/frontend/src/components/JWTDebug.vue (hunk 4 above)
Comment (jujaga) [reply app/frontend/src/components/JWTDebug.vue]: Had an internal discussion about this - leaving this for now as this is the only component checking for development mode status. If we need more devMode things, then we will need to consider either a slot pattern (similar to authenticated) or moving the v-if logic up the component chain.

**Pull request merged**